sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthenticationRecord.java

```diff
@@ -96,23 +101,43 @@ public String getUsername() {
      * @return A {@link Mono} containing {@link Void}
      */
     public Mono<OutputStream> serializeAsync(OutputStream outputStream) {
-        return Mono.defer(() -> {
-            try {
-                OBJECT_MAPPER.writeValue(outputStream, this);
-            } catch (IOException e) {
-                return Mono.error(e);
-            }
-            return Mono.just(outputStream);
+        return Mono.fromCallable(() -> {
+            serialize(outputStream);
+            return outputStream;
         });
     }
 
     /**
      * Serializes the {@link AuthenticationRecord} to the specified {@link OutputStream}
      *
      * @param outputStream The {@link OutputStream} to which the serialized record will be written to.
+     * @throws UncheckedIOException If an I/O error occurs while deserializing.
      */
     public void serialize(OutputStream outputStream) {
-        serializeAsync(outputStream).block();
+        try (JsonGenerator generator = JSON_FACTORY.createGenerator(outputStream)) {
+            if (authority != null) {
+                generator.writeStringField("authority", authority);
+            }
+
+            if (homeAccountId != null) {
+                generator.writeStringField("homeAccountId", homeAccountId);
+            }
+
+            if (tenantId != null) {
+                generator.writeStringField("tenantId", tenantId);
+            }
+
+            if (username != null) {
+                generator.writeStringField("username", username);
+            }
+
+            if (clientId != null) {
+                generator.writeStringField("clientId", clientId);
+            }
+
```

Comment:
It'll be great to explore convenience methods to reduce the amount of verbosity, especially given this will be a common pattern.